message/FieldMessageSerializer.cpp

```diff
@@ -47,6 +52,7 @@ FieldMessage FieldMessageSerializer::Deserialize(const std::string& serialized)
             int8_t fieldType;
             // 127 or 0
             iss.read(reinterpret_cast<char *>(&fieldType), sizeof(fieldType));
+            assert(fieldType == INT32_TYPE_CODE || fieldType == STRING_TYPE_CODE);
```

Comment:
`assert` only works when debug. I think it's more useful for such cases:
```cxx
// in std::span
auto operator[](size_t p) -> T& {
    assert(p < size()); // in debug it will prevent UB and show the error 
    return ...;
}
```
In this case, the serialization may not work correctly if we do not consider these cases exceptional